public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts

```diff
@@ -51,6 +59,47 @@ export default class AzureMonitorDatasource extends DataSourceWithBackend<AzureM
     );
   }
 
+  query(request: DataQueryRequest<AzureMonitorQuery>): Observable<DataQueryResponse> {
+    const metricQueries = request.targets.reduce((prev: Record<string, AzureMonitorQuery>, cur) => {
+      prev[cur.refId] = cur;
+      return prev;
+    }, {});
+
+    return super.query(request).pipe(
+      mergeMap((res: DataQueryResponse) => {
+        return from(this.processResponse(res, metricQueries));
+      })
+    );
+  }
+
+  async processResponse(
+    res: DataQueryResponse,
+    metricQueries: Record<string, AzureMonitorQuery>
+  ): Promise<DataQueryResponse> {
+    if (res.data) {
+      for (const df of res.data) {
+        const metricQuery = metricQueries[df.refId]?.azureMonitor;
+        if (metricQuery) {
+          const url =
```

Comment:
Is this the correct URL? It seems to force me to the ms.portal.azure.com "Preview" environment and Microsoft account, rather than my Grafana account where the resource actually is. The format (domain, and the `#@microsoft.onmicrosoft.com`) seems to be different to the log analytics link in https://github.com/grafana/grafana/blob/01cef3435ebad2646aa952fc821a7652af414ca6/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts#L191-L198

Reply:
Fixed the url.